### PR TITLE
[YUNIKORN-1468] Remove completedApplications from Queue

### PR DESCRIPTION
**Pull request by bgrams**
### What is this PR for?
As described in the Jira, this is a more comprehensive fix for the issues identified via YUNIKORN-1440 and discussed in #471. Proposed changes ensure that app history and cleanup mechanisms are solely managed at the Partition level and unlinked from the Queue. Queue history is also removed from the `/ws/v1/partition/{partition}/queue/{queue}/applications` REST endpoint.

### What type of PR is it?
* [x] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-1468

### How should this be tested?
Tests modified but largely unchanged.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/webservice/handlers.go [hunk 1]
```diff
@@ -534,15 +534,18 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
...
 	}
-	for _, app := range completedApps {
-		appsDao = append(appsDao, getApplicationJSON(app))
+
+	for _, app := range partitionContext.GetCompletedApplications() {
+		if app.GetQueuePath() == queueName {
+			appsDao = append(appsDao, getApplicationJSON(app))
+		}
 	}
+
```
Comment (wilfred-s) [inline pkg/webservice/handlers.go:548]: I would suggest that we remove that and add a release note for the same to YuniKorn 1.2 explaining the reasoning behind the change.

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: pkg/webservice/handlers.go (hunk 1 above)
Comment (craigcondit) [reply pkg/webservice/handlers.go:548]: Agreed. We can deprecate the field in the 1.2 docs, and remove in 2.0. It will just be empty until then.

**Review by craigcondit — CHANGES_REQUESTED** [review]
Looks good, we should just remove the iteration through apps for the queue REST API.

File: pkg/webservice/handlers.go (hunk 1 above)
Comment (bgrams) [reply pkg/webservice/handlers.go]: SGTM. Addressed in latest commits.

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**